### ISSUE TITLE
FallThroughCheck doesn't handle a synchronized block that definitely terminates

### ISSUE THREAD

**Issue by cgdecker** [checkstyle/checkstyle#2594]
It probably should.

**Comment by mkordas** [comment]
@cgdecker 
Can you provide some code example with false positive?

**Comment by romani** [comment]
please do not forget to provide configuration for this Check that you use

**Comment by romani** [comment]
Check documentation - http://checkstyle.sourceforge.net/config_coding.html#FallThrough

http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

can not reproduce a problem:
```
/var/tmp $ cat Test.java 
class Test {
	void foo () {
		switch(val)
		{
		    case 1:
		        synchronized(someLock)
		        {
		            doSomething();
		        }
    		    //fallthrough
		    case 2:
		        doSomethingElse();
		        //fallthrough
		    case 3:
		        doSomethingElse();
		    default:   /// line 16, warning is here only, as expected !!!
		        doOtherStuff();
		}
	}

/var/tmp $ cat Test.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
	    <module name="FallThrough" />
    </module>
</module>

/var/tmp $ java -jar checkstyle-6.13-all.jar -c Test.xml Test.java
Starting audit...
/var/tmp/Test.java:16:21: error: Fall through from previous branch of the switch statement.
Audit done.
Checkstyle ends with 1 errors.
```
